src/core/network/webhook.ts

```diff
@@ -5,46 +5,39 @@ const debug = d('telegraf:webhook')
 
 export default function (
   hookPath: string,
-  updateHandler: (update: Update, res: http.ServerResponse) => Promise<void>,
-  errorHandler: (err: SyntaxError) => void
+  updateHandler: (update: Update, res: http.ServerResponse) => Promise<void>
 ) {
-  return (
+  return async (
     req: http.IncomingMessage,
     res: http.ServerResponse,
-    next?: () => void
-  ): void => {
-    debug('Incoming request', req.method, req.url)
-    if (req.method !== 'POST' || req.url !== hookPath) {
-      if (typeof next === 'function') {
-        return next()
-      }
+    next = (): void => {
       res.statusCode = 403
       return res.end()
     }
+  ): Promise<void> => {
+    debug('Incoming request', req.method, req.url)
+    if (req.method !== 'POST' || req.url !== hookPath) {
+      return next()
+    }
     let body = ''
-    req.on('data', (chunk: string) => {
-      body += chunk.toString()
-    })
-    req.on('end', () => {
-      let update: Update
-      try {
-        update = JSON.parse(body)
-      } catch (error) {
-        res.writeHead(415)
-        res.end()
-        return errorHandler(error)
-      }
-      updateHandler(update, res)
-        .then(() => {
-          if (!res.finished) {
-            res.end()
-          }
-        })
```

Comment:
Previously, if `updateHandler` completed successfully, `res.end()` was called if necessary. Is there a reason why you chose not to do this anymore?

Reply:
yourself removed in the remove duplicate res.end commit

https://github.com/telegraf/telegraf/pull/1268/commits/226cbfbba14ad43cf2646875c425174ce40e1661

Reply:
@KnorpelSenf 
i forgot to mention you in previous message
this line removed by @wojpawlik in this commit https://github.com/telegraf/telegraf/commit/226cbfbba14ad43cf2646875c425174ce40e1661

Reply:
https://github.com/telegraf/telegraf/pull/1268#discussion_r550492121